mitosheet/src/jupyter/notebook/extensionUtils.tsx

```diff
@@ -228,13 +228,16 @@ export const notebookWriteGeneratedCodeToCell = (analysisName: string, codeLines
     const activeCellIndex = (window as any).Jupyter?.notebook?.get_anchor_index() || 0;
 
     const codeCell = getCellAtIndex(mitosheetCallIndex + 1);
+    const codeCellText = getCellText(codeCell);
 
-    if (isEmptyCell(codeCell) || containsGeneratedCodeOfAnalysis(getCellText(codeCell), analysisName)) {
-        if (!isEmptyCell(codeCell) && hasCodeCellBeenEditedByUser(oldCode, codeCell)) {
-            return;
-        }
+    if ((overwriteIfUserEditedCode || !hasCodeCellBeenEditedByUser(oldCode, codeCellText)) && (isEmptyCell(codeCell) || containsGeneratedCodeOfAnalysis(getCellText(codeCell), analysisName))) {
         writeToCell(codeCell, code)
     } else {
+        // Prevent overwriting the cell if the user has changed the code
+        if (overwriteIfUserEditedCode === undefined && !isEmptyCell(codeCell) && hasCodeCellBeenEditedByUser(oldCode, codeCellText)) {
+            triggerUserEditedCodeDialog();
+            return;
+        }
```

Comment:
Ok! I actually realized that the second half of the if statement on line 233 (after the &&) is unnecessary because we're already checking the code, so I'll also simplify that one